### PR TITLE
fix: TT-203 Customers Page: review the actions to ResetProjectToEdit …

### PR DESCRIPTION
**Pull request by LEON12699**
- ProjectType form and Project Form reset when discard the changes in the modal to alert "You have changes "

### CODE REVIEW

File: src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.spec.ts [hunk 1]
```diff
@@ -121,6 +122,13 @@ describe('InputProjectTypeComponent', () => {
     expect(store.dispatch).toHaveBeenCalledWith(new CreateProjectType(projectTypeData));
   });
 
+  it('should reset Project Type Form if projectTypeIdToEdit is null', () => {
+    spyOn(component.projectTypeForm, 'reset');
+    projectTypeIdToEditMock = store.overrideSelector(projectTypeIdToEdit, null);
+    store.refreshState();
+    expect(component.projectTypeForm.reset).toHaveBeenCalled();
```
Comment (scastillo-jp) [inline src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.spec.ts:129]: add a blank line before and after SpyOn.

File: src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.spec.ts [hunk 2]
```diff
@@ -121,6 +122,13 @@ describe('InputProjectTypeComponent', () => {
     expect(store.dispatch).toHaveBeenCalledWith(new CreateProjectType(projectTypeData));
   });
 
+  it('should reset Project Type Form if projectTypeIdToEdit is null', () => {
+    spyOn(component.projectTypeForm, 'reset');
+    projectTypeIdToEditMock = store.overrideSelector(projectTypeIdToEdit, null);
```
Comment (scastillo-jp) [inline src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.spec.ts:127]: what is this variable "projectTypeIdToEditMock" used for?

File: src/app/modules/customer-management/components/projects-type/store/project-type.selectors.ts [hunk 3]
```diff
@@ -5,11 +5,11 @@ import { ProjectTypeState } from './project-type.reducers';
 const getProjectTypeState = createFeatureSelector<ProjectTypeState>('projectType');
 
 export const allProjectTypes = createSelector(getProjectTypeState, (state: ProjectTypeState) => {
-  return state.data;
+  return state?.data;
```
Comment (scastillo-jp) [inline src/app/modules/customer-management/components/projects-type/store/project-type.selectors.ts:8]: If it has no data, what does it return? a null

File: src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.spec.ts (hunk 2 above)
Comment (LEON12699) [reply src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.spec.ts:127]: It is not necessary I will remove it

**Review by scastillo-jp — APPROVED** [review]

**Pull request merged**